backend/tracim_backend/lib/collaborative_document_edition/collaboration_document_edition.py

```diff
@@ -0,0 +1,124 @@
+                    )
+                )
+        return supported_collabora_file
+
+    def get_collaboration_document_edition_config(self) -> CollaborativeDocumentEditionConfig:
+        return CollaborativeDocumentEditionConfig(
+            software=self._config.COLLABORATIVE_DOCUMENT_EDITION__SOFTWARE,
+            supported_file_types=self.discover(),
+        )
+
+    def get_file_template_list(self) -> FileTemplateList:
+        return FileTemplateList(file_templates=self._get_template_list())
+
+    def _has_template_extension(self, template_name: str, extension_list: typing.List[str]):
+        for extension in extension_list:
+            if template_name.endswith(extension):
+                return True
+        return False
+
+    def _get_template_list(self) -> typing.List[str]:
+        template_list = []
+        try:
+            is_dir_exist(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_dir_readable(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+        except (NotADirectoryError) as exc:
+            raise FileTemplateNotAvailable from exc
+        for filename in os.listdir(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR):
+            if not isfile(
+                join(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR, filename)
+            ):
+                continue
+            template_list.append(filename)
+        return template_list
+
+    def _get_file_template_path(self, template_filename) -> str:
+        template_path = "{}/{}".format(
+            self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR, template_filename
+        )
+        try:
+            is_dir_exist(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_dir_readable(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR)
+            is_file_exist(template_path)
+            is_file_readable(template_path)
+        except (NotADirectoryError, NotAFileError, NotReadableFile) as exc:
+            raise FileTemplateNotAvailable from exc
+        return template_path
+
+    def check_template_available(self, template_filename) -> None:
+        self._get_file_template_path(template_filename)
+
+    def update_from_template(self, content: Content, template_filename: str) -> Content:
+        template_path = self._get_file_template_path(template_filename)
+        new_mimetype = mimetypes.guess_type(template_path)[0]
+        api = ContentApi(config=self._config, session=self._session, current_user=self._user)
+        with open(self._get_file_template_path(template_filename), "rb") as file:
+            api.update_file_data(
+                content, new_filename=content.file_name, new_mimetype=new_mimetype, new_content=file
+            )
+        return content
+
+    def get_token(self, access_token: str) -> CollaborativeDocumentEditionToken:
+        return CollaborativeDocumentEditionToken(access_token=access_token)
```

Comment:
not clear the distinction between access_token and the returned token? need a docstring

Reply:
done.